### PR TITLE
New assertRedirectedTo assertion for HTTP testing.

### PR DESCRIPTION
**Pull request by lonnieezell**
Adds a new assertion for HTTP testing: `assertRedirectedTo()` that combines `assertRedirect()`while comparing the passed in URI with the `getRedirectURL()` value.

### CODE REVIEW

File: system/Test/FeatureResponse.php [hunk 1]
```diff
@@ -98,6 +98,27 @@ public function assertRedirect()
 		$this->assertTrue($this->isRedirect(), 'Response is not a redirect or RedirectResponse.');
 	}
 
+	/**
+	 * Assert that a given response was a redirect
+	 * and it was redirect to a specific URI.
+	 *
+	 * @param string $uri
+	 */
+	public function assertRedirectedTo(string $uri)
```
Comment (MGatner) [inline system/Test/FeatureResponse.php:107]: > and it was redirect

> assertRedirectedTo

I think we should choose a case and stick with it. I would prefer "redirect" to be consistent with the other methods (so `assertRedirectTo()`) but as long as it is consistent and clear.

File: system/Test/FeatureResponse.php [hunk 2]
```diff
@@ -98,6 +98,27 @@ public function assertRedirect()
...
+	{
+		if (! $this->isRedirect())
+		{
+			return false;
+		}
+
+		$uri         = trim(strtolower($uri));
```
Comment (MGatner) [inline system/Test/FeatureResponse.php:114]: Why lower case? I don't think the return from `getRedirectUrl()` will necessarily be lower.

File: tests/system/Test/FeatureResponseTest.php [hunk 3]
```diff
@@ -151,6 +151,15 @@ public function testGetRedirectUrlReturnsNull()
...
 
+	public function testRedirectedToSuccess()
+	{
+		$this->getFeatureResponse('<h1>Hello World</h1>');
+		$this->feature->response = new RedirectResponse(new App());
+		$this->feature->response->redirect('foo/bar');
+
+		$this->feature->assertRedirectedTo('foo/bar');
+	}
+
```
Comment (MGatner) [inline tests/system/Test/FeatureResponseTest.php:162]: I think a test using a full URL is in order since your method supports it.

**Review by MGatner — CHANGES_REQUESTED** [review]
This is a great addition! I basically end up doing this myself in almost every project 🙄
I added some thoughts.

File: system/Test/FeatureResponse.php (hunk 2 above)
Comment (lonnieezell) [reply system/Test/FeatureResponse.php:114]: might be true. I know headers are case-insensitive, which obviously isn't the same. Will turn both to lowercase then.

**Comment by lonnieezell** [comment]
Changes made.

File: user_guide_src/source/testing/feature.rst [hunk 4]
```diff
@@ -198,6 +198,14 @@ Asserts that the Response is an instance of RedirectResponse.
 
     $result->assertRedirect();
 
+**assertRedirectedTo()**
```
Comment (paulbalandan) [inline user_guide_src/source/testing/feature.rst:201]: docs is not yet updated

Comment (kenjis) [reply user_guide_src/source/testing/feature.rst:201]: @lonnieezell Could you update and fix the conflict in changelog?

**Comment by lonnieezell** [comment]
@MGatner changes ready for re-review

**Review by MGatner — CHANGES_REQUESTED** [review]

**Comment by lonnieezell** [comment]
@MGatner updated

**Comment by MGatner** [comment]
I think this is ready but needs a rebase after the recent Rector fixes.

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
Glad to see this make it in! Precisely the area I am working on.

**Pull request merged**